shell/common/platform_view.h

```diff
@@ -25,95 +25,507 @@ namespace flutter {
+  ///             platform specific functions. However, they must call the base
+  ///             class method at some point in their implementation.
+  ///
   virtual void NotifyDestroyed();
 
-  // Unlike all other methods on the platform view, this one may be called on a
-  // non-platform task runner.
+  //----------------------------------------------------------------------------
+  /// @brief      Used by the shell to obtain a Skia GPU context that is capable
+  ///             of operating on the IO thread. The context must be in the same
+  ///             sharegroup as the Skia GPU context used on the render thread.
+  ///             This context will always be used on the IO thread. Because it
+  ///             is in the same sharegroup as the separate render thread
+  ///             context, any GPU resources uploaded in this context will be
+  ///             visible to the render thread context (synchronization of GPU
+  ///             resources is managed by Skia).
+  ///
+  ///             If such context cannot be created on the IO thread, callers
+  ///             may return `nullptr`. This will mean that all texture uploads
+  ///             will be queued onto the render thread which will cause
+  ///             performance issues. When this context is `nullptr`, an error
+  ///             is logged to the console. It is highly recommended that all
+  ///             platforms provide a resource context.
+  ///
+  /// @attention  Unlike all other methods on the platform view, this will be
+  ///             called on IO task runner.
+  ///
+  /// @return     The Skia GPU context that is in the same sharegroup as the
+  ///             main render thread GPU context. May be nullptr in case such a
+  ///             context cannot be created.
+  ///
   virtual sk_sp<GrContext> CreateResourceContext() const;
 
-  // Unlike all other methods on the platform view, this one may be called on a
-  // non-platform task runner.
+  //----------------------------------------------------------------------------
+  /// @brief      Used by the shell to notify the embedder that the resource
+  ///             context previously obtained via a call to
+  ///             `CreateResourceContext()` is being collected. The embedder is
+  ///             free to collect an platform specific resources associated with
+  ///             this context.
+  ///
+  /// @attention  Unlike all other methods on the platform view, this will be
+  ///             called on IO task runner.
+  ///
   virtual void ReleaseResourceContext() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Returns a weak pointer to the platform view. Since the
+  ///             platform view may only be created, accessed and destroyed on
+  ///             the platform thread, any access to the platform view from a
+  ///             non-platform task runner need a weak pointer to the platform
```

Comment:
nit: needs a weak pointer (or 'accesses' on the previous line)

Reply:
Done.